src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+#if DEBUG
+using System.Runtime.CompilerServices;
+#endif
+
+using SafeCurlHandle = Interop.libcurl.SafeCurlHandle;
+using curlioerr = Interop.libcurl.curlioerr;
+using curliocmd = Interop.libcurl.curliocmd;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler
+    {
+        private static readonly char[] s_LineEndings = new char[] { HttpRuleParser.CR, HttpRuleParser.LF };
+        private static readonly string s_LineEndingDelimiter = "\r\n";
+        private ulong CurlReceiveHeadersCallback(
+            IntPtr buffer,
+            ulong size,
+            ulong nitems,
+            IntPtr context)
+        {
+            return ExecuteCallback(buffer, size, nitems, context,
+                (ptr, totalSize, state) =>
+                {
+                    state.CancellationToken.ThrowIfCancellationRequested();
+
+#if DEBUG
+                    LogCurlBuffer(ptr, totalSize);
+#endif
+
+                    // The callback is invoked once per header and multi-line
+                    // headers get merged into a single line
+                    string responseHeader = Marshal.PtrToStringAnsi(ptr).Trim(s_LineEndings).Trim();
```

Comment:
What's the second trim here for?  All of the line endings are considered whitespace, so isn't the second trim sufficient?